codec/common/src/cpu.cpp

```diff
@@ -309,12 +309,37 @@ uint32_t WelsCPUFeatureDetect (int32_t* pNumberOfLogicProcessors) {
 
 #elif defined(mips)
 /* for loongson */
+static uint32_t get_cpu_flags_from_cpuinfo(void)
+{
+    uint32_t flags = 0;
+
+# ifdef __linux__
+    FILE* fp = fopen("/proc/cpuinfo", "r");
+    if (!fp)
+        return flags;
+
+    char buf[200];
+    memset(buf, 0, sizeof(buf));
+    while (fgets(buf, sizeof(buf), fp)) {
+        if (!strncmp(buf, "model name", strlen("model name"))) {
+            if (strstr(buf, "3A4000")) {
+                flags |= WELS_CPU_MSA | WELS_CPU_MMI;
+            } else if (strstr(buf, "2K1000")) {
+                flags |= WELS_CPU_MSA | WELS_CPU_MMI;
+            } else if (strstr(buf, "Loongson-3A") || strstr(buf, "Loongson-3B")) {
+                flags |= WELS_CPU_MMI;
+            }
+            break;
+        }
+    }
```

Comment:
We'd better also check MMI in ASEs implemented here.
Kernels from Lemote didn't display model as Loongson-3A/Loongson-3B but they'll provide loongson-mmi and loongson-ext in ASEs Implemented.

Reply:
Thank you for your advice, I will check model name and ASEs for mmi, check ASEs for msa.